cpp/src/arrow/compute/api_scalar.cc

```diff
@@ -934,6 +934,12 @@ Result<Datum> MapLookup(const Datum& arg, MapLookupOptions options, ExecContext*
   return CallFunction("map_lookup", {arg}, &options, ctx);
 }
 
+// ----------------------------------------------------------------------
+// Hash functions
+Result<Datum> Hash64(const Datum& input_array, ExecContext* ctx) {
+  return CallFunction("hash_64", {input_array}, ctx);
```

Comment:
Should we consider naming it `hash64` instead of `hash_64`?

Reply:
Yes, I am in favor of that.